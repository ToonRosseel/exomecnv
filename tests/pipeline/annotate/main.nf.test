nextflow_pipeline {

    name "Test pipeline - VEP annotation"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_annotate"

    options "-stub"

    test("pipeline - exomedepth (CRAM input) + annotate") {

        when {
            params {
                outdir = "$outputDir"
                exomedepth = true
                annotate = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, includeDir: true, include: ['*', '**/*'], ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, includeDir: false, include: ['*', '**/*'], ignore: ['exomedepth/cnv_call_vep/*vep*'], ignoreFile: 'tests/.nftignore')
            // stable_path: VEP VCF files
            def stable_path_vcf = getAllFilesFromDir(params.outdir, includeDir: false, include: ['exomedepth/cnv_call_vep/*vep.vcf.gz}'], ignore: ['*', '**/*'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    getRelativePath(stable_name, outputDir),
                    // All files with stable contents
                    stable_path,
                    // VCF files with stable contents
                    stable_path_vcf.collect { ["VCF_annotated", path(it.toString()).vcf.variantsMD5] }

                ).match("exomedepth_cram + annotate") }
            )
        }
    }
}
