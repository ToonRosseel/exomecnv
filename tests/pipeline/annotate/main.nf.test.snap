{
    "exomedepth_bed + annotate": {
        "content": [
            73,
            {
                "BCFTOOLS_SORT": {
                    "bcftools": 1.21
                },
                "BEDGOVCF": {
                    "bedgovcf": "0.1.0"
                },
                "BEDTOOLS_MAP": {
                    "bedtools": "2.31.1"
                },
                "CUSTOM_MERGECOUNTS": {
                    "paste": 9.3,
                    "mv": 9.3,
                    "cp": 9.3
                },
                "CUSTOM_REFORMATCOUNTS": {
                    "echo": "--version",
                    "cut": "cp (GNU coreutils) 9.3"
                },
                "EXOMEDEPTH_CALL": {
                    "ExomeDepth": "1.1.18",
                    "R": "Rscript (R) version 4.4.3 (2025-02-28)"
                },
                "Workflow": {
                    "nf-cmgg/exomecnv": "v2.0.0"
                }
            },
            [
                "exomedepth",
                "exomedepth/cnv_call",
                "exomedepth/cnv_call/Sample1.vcf.gz",
                "exomedepth/cnv_call/Sample1.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample10.vcf.gz",
                "exomedepth/cnv_call/Sample10.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample2.vcf.gz",
                "exomedepth/cnv_call/Sample2.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample3.vcf.gz",
                "exomedepth/cnv_call/Sample3.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample4.vcf.gz",
                "exomedepth/cnv_call/Sample4.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample5.vcf.gz",
                "exomedepth/cnv_call/Sample5.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample6.vcf.gz",
                "exomedepth/cnv_call/Sample6.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample7.vcf.gz",
                "exomedepth/cnv_call/Sample7.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample8.vcf.gz",
                "exomedepth/cnv_call/Sample8.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample9.vcf.gz",
                "exomedepth/cnv_call/Sample9.vcf.gz.tbi",
                "exomedepth/cnv_call_vep",
                "exomedepth/cnv_call_vep/Sample1.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample1.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample1.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample1.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample10.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample10.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample10.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample10.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample2.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample2.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample2.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample2.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample3.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample3.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample3.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample3.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample4.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample4.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample4.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample4.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample5.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample5.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample5.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample5.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample6.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample6.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample6.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample6.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample7.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample7.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample7.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample7.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample8.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample8.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample8.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample8.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample9.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample9.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample9.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample9.vep_summary.html",
                "exomedepth/counts",
                "exomedepth/counts/prep_F.merged.txt",
                "exomedepth/counts/prep_M.merged.txt",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_pipeline_software_mqc_versions.yml"
            ],
            [
                "Sample1.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample1.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample10.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample10.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample2.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample2.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample3.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample3.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample4.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample4.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample5.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample5.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample6.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample6.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample7.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample7.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample8.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample8.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample9.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample9.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "prep_F.merged.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "prep_M.merged.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-09-03T09:11:34.289132891"
    },
    "exomedepth_cram + annotate": {
        "content": [
            83,
            {
                "BCFTOOLS_SORT": {
                    "bcftools": 1.21
                },
                "BEDGOVCF": {
                    "bedgovcf": "0.1.0"
                },
                "BEDTOOLS_MAP": {
                    "bedtools": "2.31.1"
                },
                "CUSTOM_MERGECOUNTS": {
                    "paste": 9.3,
                    "mv": 9.3,
                    "cp": 9.3
                },
                "CUSTOM_REFORMATCOUNTS": {
                    "echo": "--version",
                    "cut": "cp (GNU coreutils) 9.3"
                },
                "EXOMEDEPTH_CALL": {
                    "ExomeDepth": "1.1.18",
                    "R": "Rscript (R) version 4.4.3 (2025-02-28)"
                },
                "MOSDEPTH": {
                    "mosdepth": "0.3.10"
                },
                "Workflow": {
                    "nf-cmgg/exomecnv": "v2.0.0"
                }
            },
            [
                "exomedepth",
                "exomedepth/cnv_call",
                "exomedepth/cnv_call/Sample1.vcf.gz",
                "exomedepth/cnv_call/Sample1.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample10.vcf.gz",
                "exomedepth/cnv_call/Sample10.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample2.vcf.gz",
                "exomedepth/cnv_call/Sample2.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample3.vcf.gz",
                "exomedepth/cnv_call/Sample3.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample4.vcf.gz",
                "exomedepth/cnv_call/Sample4.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample5.vcf.gz",
                "exomedepth/cnv_call/Sample5.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample6.vcf.gz",
                "exomedepth/cnv_call/Sample6.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample7.vcf.gz",
                "exomedepth/cnv_call/Sample7.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample8.vcf.gz",
                "exomedepth/cnv_call/Sample8.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample9.vcf.gz",
                "exomedepth/cnv_call/Sample9.vcf.gz.tbi",
                "exomedepth/cnv_call_vep",
                "exomedepth/cnv_call_vep/Sample1.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample1.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample1.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample1.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample10.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample10.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample10.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample10.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample2.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample2.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample2.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample2.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample3.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample3.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample3.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample3.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample4.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample4.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample4.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample4.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample5.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample5.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample5.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample5.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample6.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample6.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample6.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample6.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample7.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample7.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample7.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample7.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample8.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample8.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample8.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample8.vep_summary.html",
                "exomedepth/cnv_call_vep/Sample9.vep.vcf.gz",
                "exomedepth/cnv_call_vep/Sample9.vep.vcf.gz.csi",
                "exomedepth/cnv_call_vep/Sample9.vep.vcf.gz.tbi",
                "exomedepth/cnv_call_vep/Sample9.vep_summary.html",
                "exomedepth/counts",
                "exomedepth/counts/prep_F.merged.txt",
                "exomedepth/counts/prep_M.merged.txt",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_pipeline_software_mqc_versions.yml"
            ],
            [
                "Sample1.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample1.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample10.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample10.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample2.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample2.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample3.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample3.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample4.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample4.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample5.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample5.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample6.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample6.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample7.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample7.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample8.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample8.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Sample9.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "Sample9.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "prep_F.merged.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "prep_M.merged.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-09-01T16:47:37.156678284"
    }
}